examples/teleport-usage/teleport_usage.py

```diff
@@ -0,0 +1,123 @@
+"""
+Copyright 2023 Gravitational, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import json
+import datetime
+import boto3
+import os
+
+from boto3.dynamodb.conditions import Key, Attr
+
+
+# Check for AWS DynamoDB table name and region ENV, if none prompt
+TABLE_NAME = os.getenv("TABLE_NAME")
+if not TABLE_NAME:
+    TABLE_NAME = input("DynamoDB events table name: ")
+
+aws_region = os.getenv("AWS_REGION")
+if not aws_region:
+    aws_region = input("AWS Region for DynamoDB: ")
+
+# Check ENV for start and end times, if none, set last 30 days
+start_date = os.getenv("START_DATE")
+if not start_date:
+    now = datetime.datetime.now()
+    start_date = now - datetime.timedelta(days=30)
+else:
+    start_date = datetime.datetime.strptime(start_date,'%Y-%m-%d')    
+
+end_date = start_date + datetime.timedelta(days=30)
+
+# Configure DynamoDB for query
+dynamodb = boto3.resource("dynamodb", region_name=aws_region)
+table = dynamodb.Table(TABLE_NAME)
+
+# variables for protocols
+app = 'app.session.start'
+db = 'db.session.start'
+kube = 'kube.request'
+ssh = 'session.start'
+win = 'windows.desktop.session.start'
+
+protocols = ['app', 'db', 'kube', 'win']

```

Comment:
A comment as to why `ssh` is omitted here would be helpful.

Reply:
When you run the ssh together, since it has `session.start` the same as the other protocols, it does not perform a proper query with the filter and causes it to return all db and app sessions under ssh. Super weird bug even when running the `eq` parameter. The fix was to separate SSH and seems to work. @GavinFrazar also commented on this and you can see on a previous thread he started.

Reply:
Right, I would just add this info to the code so it doesn't get lost.

Reply:
Oh, yes makes sense. I will add a comment.

Reply:
Still awaiting this comment.

Reply:
No longer relevant. Refactored the script and it is no longer separate.